go_metrics/metrics/tests/test_graphite_time_parser.py

```diff
@@ -0,0 +1,168 @@
+from datetime import datetime
+
+from twisted.trial.unittest import TestCase
+
+from go_metrics.metrics.graphite_time_parser import (
+    TimeParserValueError, interval_to_seconds, parse_time)
+
+
+class TestGraphiteTimeParser(TestCase):
+    """
+    Tests for Graphite-compatible time parsing functions.
+    """
+
+    def assert_TPVE(self, *args, **kw):
+        """
+        Convenience wrapper for asserting TimeParserValueError is raised.
+        """
+        return self.assertRaises(TimeParserValueError, *args, **kw)
+
+    def assert_interval_to_seconds(self, expected, *inputs):
+        """
+        Convenience wrapper for testing multiple equivalent interval strings at
+        once.
+        """
+        for value in inputs:
+            self.assertEqual(interval_to_seconds(value), expected)
+
+    def test_interval_to_seconds_with_invalid_value(self):
+        """
+        A TimeParserValueError is raised for various invalid values.
+        """
+        self.assert_TPVE(interval_to_seconds, "")
+        self.assert_TPVE(interval_to_seconds, "s")
+        self.assert_TPVE(interval_to_seconds, "1")
+        self.assert_TPVE(interval_to_seconds, "three days")
+        self.assert_TPVE(interval_to_seconds, "-1h")
+        self.assert_TPVE(interval_to_seconds, "2fortnights")
```

Comment:
Would it be worth adding an assertion for something like `2mi`? Doesn't seem very necessary, just thought it would be nice to explicitly show that we differentiate from graphite.